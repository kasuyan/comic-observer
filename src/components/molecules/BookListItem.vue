<template>
  <li>
    <BookImage :src="data.image" :width="80" :height="108" :alt="data.name" />
    <!-- <img v-if="data.image" :src="data.image" alt />
    <div v-else class="no-image">NO IMAGE</div>-->
    <h2>{{ data.name }}</h2>
  </li>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import { BookData } from "../organisms/BookList.vue";
import BookImage from "../atoms/BookImage.vue";

@Component({
  components: {
    BookImage
  }
})
export default class BookListItem extends Vue {
  @Prop() private data!: BookData;
}
</script>

<style scoped>
li {
  list-style: none;
  margin: 0;
  display: flex;
  align-items: center;
}

h2 {
  flex-basis: auto;
  flex: 1;
  font-size: 1rem;
  margin: 0 1rem 0;
}
</style>
