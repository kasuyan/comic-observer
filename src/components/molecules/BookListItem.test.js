import { shallowMount } from "@vue/test-utils";
import BookListItem from "./BookListItem.vue";

describe("BookListItem.vue", () => {
  it("BookListItem has imageUrl", () => {
    const wrapper = shallowMount(BookListItem, {
      propsData: {
        data: {
          name: "name",
          iamge: "https://iss.ndl.go.jp/thumbnail/9784091285607"
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("BookListItem does not have imageUrl", () => {
    const wrapper = shallowMount(BookListItem, {
      propsData: {
        data: {
          name: "name",
          iamge: ""
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
