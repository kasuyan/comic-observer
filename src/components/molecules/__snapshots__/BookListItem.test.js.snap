// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListItem.vue BookListItem does not have imageUrl 1`] = `
"<li>
  <bookimage-stub width=\\"80\\" height=\\"108\\" alt=\\"name\\"></bookimage-stub>
  <h2>name</h2>
</li>"
`;

exports[`BookListItem.vue BookListItem has imageUrl 1`] = `
"<li>
  <bookimage-stub width=\\"80\\" height=\\"108\\" alt=\\"name\\"></bookimage-stub>
  <h2>name</h2>
</li>"
`;
